lib/GrabViaRestPlugin/src/GrabViaRestPlugin.h

```diff
@@ -90,28 +89,18 @@ class GrabViaRestPlugin : public PluginWithConfig
         m_mutex(),
         m_isConnectionError(false),
         m_hasTopicChanged(false),
-        m_taskProxy()
+        m_restService(RestService::getInstance()),
+        m_isAllowedToSend(true)
     {
         (void)m_mutex.create();
+        m_restId = m_restService.registerPlugin();
     }

```

Comment:
If a REST id is required, allocate in the start method and release it in the stop method.